include/pybind11/chrono.h

```diff
@@ -118,6 +118,30 @@ template <typename Duration> class type_caster<std::chrono::time_point<std::chro
 
             value = system_clock::from_time_t(std::mktime(&cal)) + microseconds(PyDateTime_DATE_GET_MICROSECOND(src.ptr()));
             return true;
+        } else if (PyDate_Check(src.ptr())) {
+            std::tm cal;
```

Comment:
Could you move the std::tm object to the parent scope?

Reply:
Move all `std::tm cal;` to the line before `if (PyDateTime_Check(src.ptr())) {`?